mlflow/java/client/src/main/java/org/mlflow/artifacts/CliBasedArtifactRepository.java

```diff
@@ -140,6 +140,22 @@ public File downloadArtifacts() {
     return listArtifacts(null);
   }
 
+  /**
+   * Only available in the CliBasedArtifactRepository. Downloads an artifact to the local
+   * filesystem when provided with an artifact uri. This method should not be used directly
+   * by the user. Please use {@link org.mlflow.tracking.MlflowClient}
+   *
+   * @param artifactUri Artifact uri
+   * @return Directory/file of the artifact
+   */
+  public File downloadArtifactUri(String artifactUri) {
```

Comment:
`downloadArtifactFromUri` might be more clear (since we are not downloading the URI itself).

Reply:
Done